src/main/java/com/example/books/controller/BookController.java

```diff
@@ -30,22 +30,22 @@
 public class BookController {
     private final BookService bookService;
 
-    @PreAuthorize("hasRole('ROLE_USER')")
+    @PreAuthorize("hasAnyRole('ROLE_USER', 'ROLE_ADMIN')")

```

Comment:
Taking into account that you have only two roles I don't see necessity to put both of them into annotation.
If all roles can access endpoint you can just remove annotation as you security config requires tyo be authenticated anyway
```suggestion
```
Please recheck all similar cases